pkg/client/unversioned/clientcmd/api/v1/types.go

```diff
@@ -54,7 +54,10 @@ type Preferences struct {
 type Cluster struct {
 	// Server is the address of the kubernetes cluster (https://hostname:port).
 	Server string `json:"server"`
-	// APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	// DEPRECATED: APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	//  Because a cluster can run multiple API groups and potentially multiple versions of each, it no longer makes sense to specify
+	//  a single value for the cluster version. This field isn't really needed anyway, so we're deprecating it without a replacement.
+	//  It will be ignored if it is present.
```

Comment:
Following up on this, @caesarxuchao convinced me that there's still use for something similar, so the plan now is that @JanetKuo will replace this with a PreferredVersions setting. It will take a list of group/version pairs, and will inform version negotiation. examples:
* "v1,experimental/v1alpha1"
* "experimental/v1alpha1" - this means no preference on the v1 api version, but if you get something from experimental, you must use the v1alpha1 version.

We'll do the same for the command line flag, add a --preferred-versions flag. We'll keep --api-version flag working for requests that use a single resource.